python/llm/src/bigdl/llm/transformers/utils.py

```diff
@@ -175,3 +175,10 @@ def get_xpu_device_type(x):
         return "pvc"
     else:
         return "others"
+
+
+def quantize_kv_cache(x: torch.Tensor) -> bool:
+    if os.environ.get("BIGDL_QUANTIZE_KV_CACHE", None) is not None:
+        return os.environ["BIGDL_QUANTIZE_KV_CACHE"] == 1
+    else:
+        return x.device.type == 'xpu' and x.dtype != torch.half and x.dtype != torch.bfloat16
```

Comment:
How's the dtype of `hidden_states` determined?

Reply:
`hidden_states` is the input tensor of this layer,  its dtype is determined by `model.dtype`

Reply:
> `hidden_states` is the input tensor of this layer, its dtype is determined by `model.dtype`

`model.dtype` is different from `qtype` (i.e., the `low_in_low_bit` parameter); we probably should not quantize here if the user specifies `fp16` or `bf16` as `low_in_low_bit` when loading the model

Reply:
changed to qtype